components/higher-order/with-api-data/index.js

```diff
@@ -178,13 +180,13 @@ export default ( mapPropsToData ) => ( WrappedComponent ) => {
 
 					// Initialize pending flags as explicitly false
 					const pendingKey = this.getPendingKey( method );
-					stateValue[ pendingKey ] = false;
+					result[ propName ][ pendingKey ] = false;
 
 					// Track path for future map skipping
-					stateValue.path = path;
+					result[ propName ].path = path;
 
-					return stateValue;
-				}, {} );
+					return result[ propName ];
```

Comment:
Is there a reason for returning here? I didn't think `forEach()` did anything with the return value of the iteratee function.

Reply:
You're right, this is a remnant of the previous `Array#reduce` implementation. I'll remove.